back/engines/free/user_confirmation/app/jobs/user_confirmation/expire_confirmation_code_job.rb

```diff
@@ -2,8 +2,10 @@
 
 module UserConfirmation
   class ExpireConfirmationCodeJob < ApplicationJob
-    def run(user, code_to_expire)
-      return if user.confirmed? || user.email_confirmation_code != code_to_expire
+    def run(user_id, code_to_expire)
+      # We first check if the record exists to cover the case where the user has been deleted.
```

Comment:
Comment added, as we don't have a related test to clarify why this job is different to the others.